lib/node_main.js

```diff
@@ -13,49 +13,62 @@
 'use strict';
 
 var upgradeHtml = require('./upgrade_html');
-var yargs = require('yargs');
+var nomnom = require('nomnom')();
 var colors = require('colors/safe');
 var fs = require('fs');
 var path = require('path');
+var es6Collections = require('es6-collections');
+// jshint -W079
+var Set = es6Collections.Set || global.Set;
+// jshint +W079
 
-// NOTE(rictic): if this gets any more complicated, recommend changing to
-//     minimist or nomnom rather than introducing another args parser into
-//     the polymer team.
-var argv = yargs
-    .usage(
-      'polyup [options] ./path/to/your/v0.5/source.html\n\n' +
 
-      'Assists in upgrading code that uses Polymer from v0.5 to v1.0. Takes ' +
-      'care of most of the purely mechanical steps automatically.\n\n' +
+nomnom.script('polyup');
+nomnom.option('overwrite', {
+  flag: true,
+  help: 'Overwrite the input html and any referenced scripts on disk.'
+});
+nomnom.option('ignore', {
+  abbr: 'i',
+  list: true,
+  metavar: 'PATH',
+  default: [],
+  help: 'File(s) that polyup should not upgrade, or even try to read at all. ' +
+        ' Can be specified multiple times.'
+});
+nomnom.option('path_to_html', {
+  position: 0,
+  required: true,
+  help: 'Path to the HTML file to upgrade. The first positional argument.'
+});
+var opts = nomnom.parse();
 
-      'By default, running polyup simply prints the upgraded code to the ' +
-      'console. You must use the --overwrite option for anything to actually ' +
-      'be written to disk.')
-    .demand(1)
-    .help('help')
-    .boolean('overwrite')
-    .describe(
-      'overwrite',
-      'Overwrite the input html and any referenced scripts on disk. ' +
-      'Warning! Be confident that your source control is in a good state ' +
-      'before you override your source code!')
-    .argv;
-
-var filename = argv._[0];
+var filename = opts.path_to_html;
 if (!fs.existsSync(filename)) {
   console.error('No such file: ' + filename);
   process.exit(66);
 }
 
+var setToIgnore = new Set();
+opts.ignore.forEach(function(toIgnore) {
+  toIgnore = path.resolve(toIgnore);
+  if (!fs.existsSync(toIgnore)) {
```

Comment:
My thinking was that it's probably a typo, so you'll get unexpected behavior when the file you tried to ignore is actually overwritten.